core/commands/storage.go

```diff
@@ -1283,29 +1292,23 @@ This command updates host information and broadcasts to the BTFS network.`,
 		bl, blFound := req.Options[hostBandwidthLimitOptionName].(float64)
 		stm, stmFound := req.Options[hostStorageTimeMinOptionName].(uint64)
 
+		if sp > bttTotalSupply || cp > bttTotalSupply {
+			return fmt.Errorf("maximum price is %d", bttTotalSupply)
+		}
+
 		n, err := cmdenv.GetNode(env)
 		if err != nil {
 			return err
 		}
 
 		rds := n.Repo.Datastore()
+		peerId := n.Identity.Pretty()
 
-		selfKey := storage.GetHostStorageKey(n.Identity.Pretty())
-		b, err := rds.Get(selfKey)
-		// If key not found, create new
-		if err != nil && err != ds.ErrNotFound {
+		ns, err := GetSettings(req.Context, cfg.Services.HubDomain, peerId, rds)
+		if err != nil {
 			return err
 		}
 
-		var ns info.NodeStorage
-		if err == nil {
-			// TODO: Set default values if unset
-			err = json.Unmarshal(b, &ns)
-			if err != nil {
-				return err
-			}
-		}
-
```

Comment:
Logic is different on this change ... it should not override local settings if already set...